apps/judicial-system/backend/src/app/modules/notification/notification.service.ts

```diff
@@ -1925,7 +1966,15 @@ export class NotificationService {
               theCase,
               true,
             )
-            messages = []
+            // We still want to send the defender a link to the case even if
+            // the judge chooses not to send a calendar invitation
+            messages = [
+              this.getNotificationMessage(
+                MessageType.SEND_DEFENDER_ASSIGNED_NOTIFICATION,
+                user,
+                theCase,
+              ),
+            ]
```

Comment:
I'm not sure how I feel about doing this this way, but I didn't really have an idea of a cleaner way to do it without making the code messier or adding en entirely new event just to handle this specific case, which felt like overkill?
Feedback appreciated though. I'm a little wary of the code around notifications, it feels like it's getting harder to maintain and keep track of

Reply:
Your implementation fits well with the current notification design. I agree that notifications are getting pretty complicated. My main concern is with the granularity of notifications and consequently our ability to retry failed notifications. At the moment, we basically say that a notification succeeds if we managed to successfully send a notification to one of the participants. A more robust approach would treat notifications to individual recipients separately. This would most likely also address some of your concerns with maintainability.